write-fonts/src/tables/layout.rs

```diff
@@ -624,4 +646,33 @@ mod tests {
         assert_eq!(result.len(), 1);
         assert_eq!(result[0], 0x5540_u16);
     }
+
+    fn make_glyph_vec<const N: usize>(gids: [u16; N]) -> Vec<GlyphId> {
+        gids.into_iter().map(GlyphId::new).collect()
+    }
+
+    #[test]
+    fn coverage_builder() {
+        let coverage = make_glyph_vec([1u16, 2, 9, 3, 6, 9])
+            .into_iter()
+            .collect::<CoverageTableBuilder>();
+        assert_eq!(coverage.glyphs, make_glyph_vec([1, 2, 3, 6, 9]));
+    }
+
+    #[test]
+    fn class_def_builder_zero() {
+        // even if class 0 is provided, we don't need to assign explicit entries for it
+        fn make_class<const N: usize>(gid_class_pairs: [(u16, u16); N]) -> ClassDef {
+            gid_class_pairs
+                .iter()
+                .map(|(gid, cls)| (GlyphId::new(*gid), *cls))
+                .collect::<ClassDefBuilder>()
+                .build()
+        }
+
+        let class = make_class([(4, 0), (5, 1)]);
+        assert!(class.get_raw(GlyphId::new(4)).is_none());
+        assert_eq!(class.get_raw(GlyphId::new(5)), Some(1));
+        assert!(class.get_raw(GlyphId::new(100)).is_none());
```

Comment:
why not test the public get fn?

Reply:
The issue is I actually want to know if there is a zero value in the record, instead of it being used as a placeholder; the public fn does unwrap_or(0).